### PR TITLE
Origin/feature/savefunction booking

### PR DESCRIPTION
**Pull request by manic1990**
Added partial class for BAL
Booking view changes 
Save functionality for booking

### CODE REVIEW

File: BookingSystem/APIClient/Repository/APIClientBookingRepository.cs [hunk 1]
```diff
@@ -70,8 +70,11 @@ public async Task<BookingDataVM> CreateBookingAsync(BookingDataVM data)
                     var responses = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
                     var responseData = JsonConvert.DeserializeObject<object>(responses);
                     var bookingData = JsonConvert.DeserializeObject<ResponseModel>(responseData.ToString());
-                    var result = JsonConvert.DeserializeObject<BookingDataVM>(bookingData.response.ToString());
```
Comment (Backiaraj07) [inline BookingSystem/APIClient/Repository/APIClientBookingRepository.cs:73]: Use Try Catch block

**Review by Backiaraj07 — COMMENTED** [review]
Comments added Kindly fix this.

File: BookingSystem/Controllers/BookingController.cs [hunk 2]
```diff
@@ -99,28 +100,42 @@ public async Task<IActionResult> Booking()
...
+                var result = await _clientRepository.CreateBookingAsync(booking).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex.Message);
+            }
+            _logger.LogInformation(nameof(BookingController) + "." + nameof(Booking) + complete);
+
             return RedirectToAction("Index");
         }
```
[the remark this answers is not in the document]
Comment (manic1990) [reply BookingSystem/Controllers/BookingController.cs]: Using nameof

File: BookingSystem/Controllers/BookingController.cs [hunk 3]
```diff
@@ -99,28 +100,42 @@ public async Task<IActionResult> Booking()
...
+            var folderDetails = Path.Combine(Directory.GetCurrentDirectory(), $"Constants\\{"Json_CardTypes\\cardtypes.json"}");
+            var JSON = System.IO.File.ReadAllText(folderDetails);
+            dynamic jsonObj = Newtonsoft.Json.JsonConvert.DeserializeObject(JSON);
+            viewBookingData.CardTypes = new List<SelectListItem>();
+            foreach (var item in jsonObj.CardTypes)
+            {
+                var listItem = new SelectListItem { Value = item.Type, Text = item.Type };
+                viewBookingData.CardTypes.Add(listItem);
+            }
+
```
[the remark this answers is not in the document]
Comment (manic1990) [reply BookingSystem/Controllers/BookingController.cs]: Handled try catch block

**Review by Backiaraj07 — APPROVED** [review]
Verified

**Review by sathish199308 — APPROVED** [review]
Approved

**Pull request merged**